relay-metrics/src/bucket.rs

```diff
@@ -452,33 +517,105 @@ fn parse_timestamp(string: &str) -> Option<UnixTimestamp> {
 /// ```
 #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct Bucket {
-    /// The start time of the time window.
+    /// The start time of the bucket's time window.
     ///
     /// If a timestamp is not supplied as part of the submission payload, the default timestamp
     /// supplied to [`Bucket::parse`] or [`Bucket::parse_all`] is associated with the metric. It is
     /// then aligned with the aggregation window.
+    ///
+    /// # Statsd Format
+    ///
+    /// In statsd, timestamps are part of the `|`-separated list following values. Timestamps start
+    /// with with the literal character `'T'` followed by the UNIX timestamp.
+    ///
+    /// # Example
+    ///
+    /// ```text
+    /// endpoint.hits:1|c|T1615889440
+    /// ```
     pub timestamp: UnixTimestamp,
+
     /// The length of the time window in seconds.
+    ///
+    /// To initialize a new bucket, choose `0` as width. Once the bucket is tracked by Relay's
```

Comment:
What's relay behavior if I don't set `0`?

Reply:
It will merge the bucket based on the timestamp at the middle of the bucket's time range.